src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -109,7 +109,8 @@ ciMethod::ciMethod(const methodHandle& h_m, ciInstanceKlass* holder) :
   ciEnv *env = CURRENT_ENV;
   if (env->jvmti_can_hotswap_or_post_breakpoint()) {
     // 6328518 check hotswap conditions under the right lock.
-    MutexLocker locker(Compile_lock);
+    bool should_take_Compile_lock = !Compile_lock->owned_by_self();
+    ConditionalMutexLocker locker(Compile_lock, should_take_Compile_lock, Mutex::_safepoint_check_flag);
```

Comment:
I see no point in making an explicit variable. Most of the uses of `CML` just inline it:
https://github.com/openjdk/jdk/blob/8feabc849ba2f617c8c6dbb2ec5074297beb6437/src/hotspot/share/runtime/deoptimization.cpp#L122

Reply:
Shorter line length - fit terminal window ;^)
Also it more clear what this parameter for.